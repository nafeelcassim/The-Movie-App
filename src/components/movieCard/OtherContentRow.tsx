import React from 'react';
import {View, Text, StyleSheet} from 'react-native';
import {commonStyles} from '../../util/commonStyles';
import {appConstants} from '../../util/constants';

type OtherContentRowProps = {
  releaseData: string;
  isAdult: boolean;
  originalLanguage: string;
};

const OtherContentRow = (props: Partial<OtherContentRowProps>) => {
  const {releaseData, isAdult, originalLanguage} = props;
  return (
    <View style={styles.otherContentRow}>
      <Text style={commonStyles.otherContent}>{`${
        releaseData ?? appConstants.notAvailableString
      } (${originalLanguage ?? appConstants.notAvailableString})`}</Text>
      <View style={styles.circleView} />
      <View style={styles.adultBorder}>
        <Text adjustsFontSizeToFit={true} style={styles.adultBorderText}>
          {isAdult ? 'R' : 'All'}
        </Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  otherContentRow: {
    marginTop: 4,
    marginBottom: 4,
    flexDirection: 'row',
    alignItems: 'center',
    alignContent: 'center',
  },
  circleView: {
    height: 4,
    width: 4,
    borderRadius: 30,
    backgroundColor: 'grey',
    marginLeft: 5,
  },
  adultBorder: {
    borderWidth: 1,
    borderRadius: 3,
    borderColor: 'grey',
    padding: 1,
    marginLeft: 5,
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  adultBorderText: {
    fontSize: 11,
    color: 'grey',
  },
});

export default OtherContentRow;
