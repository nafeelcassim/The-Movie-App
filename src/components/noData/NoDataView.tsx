import React from 'react';
import {View, Text, StyleSheet} from 'react-native';

type NoDataViewProps = {
  title: string;
};

const NoDataView = (props: NoDataViewProps) => {
  const {title} = props;
  return (
    <View style={styles.container}>
      <Text>{title}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
});

export default NoDataView;
