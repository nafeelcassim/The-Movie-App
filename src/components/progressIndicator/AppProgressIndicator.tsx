import React from 'react';
import {ActivityIndicator} from 'react-native';

const AppProgressIndicator = () => {
  return <ActivityIndicator size="large" color="#002138" />;
};

export default AppProgressIndicator;
