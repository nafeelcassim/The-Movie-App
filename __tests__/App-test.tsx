/**
 * @format
 */

import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import AppProgressIndicator from '../src/components/progressIndicator/AppProgressIndicator';
import NoDataView from '../src/components/noData/NoDataView';
import OtherContentRow from '../src/components/movieCard/OtherContentRow';

// Check if App Prgress indicator renders correctly
describe('App Prgress Indicator Component', () => {
  it('renders correctly', () => {
    const progressIndicatorElement = renderer.create(<AppProgressIndicator />);
    expect(progressIndicatorElement).toBeDefined();
  });
});

// Check if NoDataView renders correctly
describe('DynamicText component', () => {
  it.each`
    input             | expected
    ${'No data view'} | ${'No data view'}
    ${'Empty view'}   | ${'Empty view'}
  `('renders $expected when given input $input', ({input}) => {
    const tree = renderer.create(<NoDataView title={input} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

// Other Content Row
describe('Other Row Content', () => {
  it('Renders the component when isAdult is false', () => {
    const tree = renderer
      .create(
        <OtherContentRow
          isAdult={false}
          originalLanguage="en"
          releaseData="2022-01-12"
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Renders the component when isAdult is true', () => {
    const tree = renderer
      .create(
        <OtherContentRow
          isAdult={true}
          originalLanguage="en"
          releaseData="2022-01-12"
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Renders the component when no parameter is passed', () => {
    const tree = renderer.create(<OtherContentRow />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
