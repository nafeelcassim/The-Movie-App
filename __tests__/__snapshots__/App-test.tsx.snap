// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamicText component renders Empty view when given input Empty view 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Empty view
  </Text>
</View>
`;

exports[`DynamicText component renders No data view when given input No data view 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    No data view
  </Text>
</View>
`;

exports[`Other Row Content Renders the component when isAdult is false 1`] = `
<View
  style={
    {
      "alignContent": "center",
      "alignItems": "center",
      "flexDirection": "row",
      "marginBottom": 4,
      "marginTop": 4,
    }
  }
>
  <Text
    style={
      {
        "color": "grey",
        "fontWeight": "300",
      }
    }
  >
    2022-01-12 (en)
  </Text>
  <View
    style={
      {
        "backgroundColor": "grey",
        "borderRadius": 30,
        "height": 4,
        "marginLeft": 5,
        "width": 4,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "borderColor": "grey",
        "borderRadius": 3,
        "borderWidth": 1,
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginLeft": 5,
        "padding": 1,
      }
    }
  >
    <Text
      adjustsFontSizeToFit={true}
      style={
        {
          "color": "grey",
          "fontSize": 11,
        }
      }
    >
      All
    </Text>
  </View>
</View>
`;

exports[`Other Row Content Renders the component when isAdult is true 1`] = `
<View
  style={
    {
      "alignContent": "center",
      "alignItems": "center",
      "flexDirection": "row",
      "marginBottom": 4,
      "marginTop": 4,
    }
  }
>
  <Text
    style={
      {
        "color": "grey",
        "fontWeight": "300",
      }
    }
  >
    2022-01-12 (en)
  </Text>
  <View
    style={
      {
        "backgroundColor": "grey",
        "borderRadius": 30,
        "height": 4,
        "marginLeft": 5,
        "width": 4,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "borderColor": "grey",
        "borderRadius": 3,
        "borderWidth": 1,
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginLeft": 5,
        "padding": 1,
      }
    }
  >
    <Text
      adjustsFontSizeToFit={true}
      style={
        {
          "color": "grey",
          "fontSize": 11,
        }
      }
    >
      R
    </Text>
  </View>
</View>
`;

exports[`Other Row Content Renders the component when no parameter is passed 1`] = `
<View
  style={
    {
      "alignContent": "center",
      "alignItems": "center",
      "flexDirection": "row",
      "marginBottom": 4,
      "marginTop": 4,
    }
  }
>
  <Text
    style={
      {
        "color": "grey",
        "fontWeight": "300",
      }
    }
  >
    N/A (N/A)
  </Text>
  <View
    style={
      {
        "backgroundColor": "grey",
        "borderRadius": 30,
        "height": 4,
        "marginLeft": 5,
        "width": 4,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "borderColor": "grey",
        "borderRadius": 3,
        "borderWidth": 1,
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginLeft": 5,
        "padding": 1,
      }
    }
  >
    <Text
      adjustsFontSizeToFit={true}
      style={
        {
          "color": "grey",
          "fontSize": 11,
        }
      }
    >
      All
    </Text>
  </View>
</View>
`;
